src/slots/Slot.js

```diff
@@ -71,18 +74,28 @@ class Slot {
     throw new Error(`${this} requires an .assertStructure method to mutate. See mixin 'JOIous'`)
   }
 
+  isUnitless() { return this.unit === Slot.UNITLESS_UNIT }
+
   /* *******************************************************************
    * GRAPH ACTIONS: CONNECTING SLOTS
    * **************************************************************** */
-  _assertConnectionBetweenIsPossible(otherSlot) {
+  _assertSlotDataTypeCompatible(otherSlot) {
     if (this.dataType !== otherSlot.dataType) {
       throw new SlotConnectionError(this, otherSlot, 'dataTypes must match between slots')
     }
+  }
 
-    if (this.unit !== otherSlot.unit) {
+  _assertSlotUnitCompatible(otherSlot) {
+    if (
+      this.unit !== otherSlot.unit
+      && !this.isUnitless()
+      && !otherSlot.isUnitless()
+    ) {
       throw new SlotConnectionError(this, otherSlot, 'units must match between slots')
     }
+  }
```

Comment:
if I have one slot that is unitless and another one that is not, should they be compatible? If I'm not mistaken, this code says yes. Do we want that behaviour or should be treat unitless just as any other "unit" by saying: only unitless slots can be connected.

Reply:
screw that, total freedom!